File: 005/Python/Problem005Tests.py
#!/usr/bin/env python

import unittest

import problem005

class Problem005Tests(unittest.TestCase):

    def testComputation(self):
        self.assertEqual(   1, problem005.smallest_multiple( 1))
        self.assertEqual(   2, problem005.smallest_multiple( 2))
        self.assertEqual(   6, problem005.smallest_multiple( 3))
        self.assertEqual(  12, problem005.smallest_multiple( 4))
        self.assertEqual(  60, problem005.smallest_multiple( 5))
        self.assertEqual(  60, problem005.smallest_multiple( 6))
        self.assertEqual(2520, problem005.smallest_multiple(10))

unittest.main()

File: 005/Erlang/problem005_tests.erl
-module(problem005_tests).

-include_lib("eunit/include/eunit.hrl").
-import(problem005, [smallest_multiple/1]).

smallest_multiple_test_() -> [
  ?_assertEqual(   1, smallest_multiple( 1)),
  ?_assertEqual(   2, smallest_multiple( 2)),
  ?_assertEqual(   6, smallest_multiple( 3)),
  ?_assertEqual(  12, smallest_multiple( 4)),
  ?_assertEqual(  60, smallest_multiple( 5)),
  ?_assertEqual(  60, smallest_multiple( 6)),
  ?_assertEqual(2520, smallest_multiple(10))
	].
